File: verve patch/base/Applications/CSymphony_0_1/vbSamples/CodeBaseReferenceExpression.vb
Option Strict Off
Option Explicit On

Imports System
Imports System.CodeDom
Imports System.CodeDom.Compiler
Imports System.Collections
Imports CSymphony.Lib

Namespace CSymphony.CSy
	Public Class CSyUnit
		 Inherits GraphUnit
		Overrides Public Sub Build()
			ns("MyNameSpace")

			declClass(TypePublic, "MyClasss", "ParentClass")
			declMethod(MemPublic, "System.Void", "MyMethod")

			Dim selfDispose As MethodRef = New MethodRef (mySuper(), "Dispose")

			enter (ex (selfDispose))

			enter(ex (New MethodRef(mySuper(), "SomeMethod")))
		End Sub
	End Class
End Namespace
			

File: verve patch/base/Applications/CSymphony_0_1/csSamples/CodeBaseReferenceExpression.cs
using System;
using System.CodeDom;
using System.CodeDom.Compiler;
using System.Collections;
using CSymphony.Lib;

namespace CSymphony.CSy
{
	public class CSyUnit : GraphUnit
	{
		public override void Build()
		{
			ns("MyNameSpace");

			declClass(TypePublic, "MyClasss", "ParentClass");
			declMethod(MemPublic, "System.Void", "MyMethod");

			MethodRef selfDispose = new MethodRef (mySuper(), "Dispose");

			enter (ex (selfDispose));

			enter(ex (new MethodRef(mySuper(), "SomeMethod")));

		}
	}
}
